app/components/sidebars/main/channels_list/list/index.js

```diff
@@ -42,6 +43,7 @@ function mapStateToProps(state) {
     const lastUnreadChannel = DeviceTypes.IS_TABLET ? state.views.channel.keepChannelIdAsUnread : null;
     const unreadChannelIds = getSortedUnreadChannelIds(state, lastUnreadChannel);
     const favoriteChannelIds = getSortedFavoriteChannelIds(state);
+    const channelsByCategory = getChannelsByCategoryForCurrentTeam(state);
```

Comment:
here you should either include this or the other ones depending on the server version if it supports categories or not.. then you avoid a lot of computation